addons/binding/org.openhab.binding.linuxinput/src/main/java/org/openhab/binding/linuxinput/internal/LinuxInputDiscoveryService.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.linuxinput.internal;
+
+import static org.openhab.binding.linuxinput.internal.LinuxInputBindingConstants.THING_TYPE_DEVICE;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.linuxinput.internal.evdev4j.EvdevDevice;
+import org.openhab.binding.linuxinput.internal.evdev4j.LastErrorException;
+import org.openhab.binding.linuxinput.internal.evdev4j.jnr.EvdevLibrary;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discovery service for LinuxInputHandlers based on the /dev/input directory.
+ *
+ * @author Thomas Weißschuh - Initial contribution
+ */
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.linuxinput")
+public class LinuxInputDiscoveryService extends AbstractDiscoveryService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LinuxInputDiscoveryService.class);
+
+    private Future<?> discoveryJob;
+    private static final Duration REFRESH_INTERVAL = Duration.ofSeconds(50);
+    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+    private static final Path DEVICE_DIRECTORY = FileSystems.getDefault().getPath("/dev/input");
+
+    public LinuxInputDiscoveryService() {
+        super(Collections.singleton(THING_TYPE_DEVICE), (int) TIMEOUT.getSeconds(), true);
+    }
+
+    @Override
+    protected void startScan() {
+        performScan(false);
+    }
+
+    private void performScan(boolean applyTtl) {
+        LOGGER.warn("startScan {}", DEVICE_DIRECTORY);
```

Comment:
In [E.4 of the coding guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging) we listed what kind of logging level should be used when. Warning should not be used here.

Reply:
Yes, the log levels are all wrong...
I will fix them.